interface.c

```diff
@@ -270,6 +272,27 @@ int check_iface(struct Interface *iface)
 			res = -1;
 		}
 
+		/* For the default route 0::/0, we need to explicitly check the
+		 * lifetime against the AdvDefaultLifetime value.
+		 *
+		 * If exactly one of the two has a zero value, then nodes processing
+		 * the RA may have a flap in their default route.
+		 *
+		 * AdvDefaultLifetime == 0 && route.AdvRouteLifetime > 0:
+		 * - default route is deleted and then re-added.
+		 * AdvDefaultLifetime > 0 && route.AdvRouteLifetime == 0:
+		 * - default route is added and then deleted.
+		 */
+		if(0 == memcmp(&(route->Prefix), &zeroaddr, sizeof(struct in6_addr))) {
+			int route_zerolife = (route->AdvRouteLifetime == 0);
+			int defaultroute_zerolife = (iface->ra_header_info.AdvDefaultLifetime == 0);
+			if( (route_zerolife && !defaultroute_zerolife) || (!route_zerolife && defaultroute_zerolife) ) {
```

Comment:
Implemented; I thought the previous version was more readable, and the compiler automatically converted to XOR, but a native XOR is fine too.